### ISSUE TITLE
Do not consume errors while (de)serializing

### ISSUE THREAD

**Issue by ChaoticTempest** [near/near-sdk-rs#558]
A user on the discord posted about some serialization issues, and it seemed to be pretty difficult to understand what was going wrong since we're consuming the errors such as in this line: [https://github.com/near/near-sdk-rs/blob/bcdf5babf9cea4aef94457767735dc9195d84798/near-sdk/src/collections/vector.rs#L183](https://github.com/near/near-sdk-rs/blob/bcdf5babf9cea4aef94457767735dc9195d84798/near-sdk/src/collections/vector.rs#L183)

A little more useful info could include which element couldn't be serialized, or maybe panicking with the whole error too. I'm not sure if the error object from this `unwrap_or_else` has any useful info though.

**Comment by austinabell** [comment]
> A user on the discord posted about some serialization issues, and it seemed to be pretty difficult to understand what was going wrong since we're consuming the errors such as in this line:
> 
> https://github.com/near/near-sdk-rs/blob/bcdf5babf9cea4aef94457767735dc9195d84798/near-sdk/src/collections/vector.rs#L183
> 
> A little more useful info could include which element couldn't be serialized, or maybe panicking with the whole error too. I'm not sure if the error object from this `unwrap_or_else` has any useful info though.

Yeah unfortunately it's a tradeoff because including this information would bloat the code size of all contracts that use it when this should never error unless either contract code is updated which changes the value serialization (while keeping existing state) or there is a collision for that index.

Possibly we should consider having some SDK feature that includes more info on errors at the cost of contract size because doing by default would negatively affect contract size for correct usage. The issue is that the error needs to be formatted to a string to panic with it when this error _shouldn't_ be hit.

**Comment by ChaoticTempest** [comment]
That's right, the formatting mechanisms would definitely bloat the code size. Even adding a feature-flag would require people to learn/discover it. Definitely not an ideal situation for debuggability if something goes wrong. But I think we can at least enable it for running sim tests (or sandbox tests in the future)
